inst/templates/ad_adcoeq.R

```diff
@@ -0,0 +1,257 @@
+# Name: ADCOEQ
+#
+# Label: Control of Eating Questionnaires Analysis dataset
+#
+# Input: adsl, qs
+
+# License ----
+
+# Note that University of Leeds are the copyright holders of the Control of Eating
+# Questionnaire (CoEQ) and the test data included within `{admiralmetabolic}` as
+# well as the ADCOEQ code are for not-for-profit use only within `{admiralmetabolic}`
+# and pharmaverse-related examples/documentation. Any persons or companies wanting
+# to use the CoEQ should request a license to do so from the following
+# [link](https://licensing.leeds.ac.uk/product/control-of-eating-questionnaire-coeq).
+
+# Attach/load required packages ----
+library(admiral)
+library(admiralmetabolic)
+library(tibble)
+library(dplyr)
+library(stringr)
+
+# Define project options/variables ----
+# Use the admiral option functionality to store subject key variables in one
+# place (note: `subject_keys` defaults to STUDYID and USUBJID)
+set_admiral_options(subject_keys = exprs(STUDYID, USUBJID))
+
+# Store ADSL variables required for derivations as an R object, enabling
+# simplified usage throughout the program
+adsl_vars <- exprs(TRTSDT, TRTEDT, TRT01P, TRT01A)
+
+# Read in data ----
+# See the "Read in Data" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#readdata)
+
+# Read data
+qs_metabolic <- admiralmetabolic::qs_metabolic
+adsl <- admiral::admiral_adsl
+
+# Convert SAS missing character values to NA
+adcoeq <- qs_metabolic %>%
+  convert_blanks_to_na()
+adsl <- adsl %>%
+  convert_blanks_to_na()
+
+# Merge ADSL variables (stored in `adsl_vars`) needed for ADQS
+adcoeq <- adcoeq %>%
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = get_admiral_option("subject_keys")
+  )
+
+# Add analysis parameter variables
+adcoeq <- adcoeq %>%
+  mutate(
+    PARAMCD = QSTESTCD,
+    PARAM = QSTEST,
+    PARCAT1 = QSCAT
+  )
+
+# Derive Date/Time and Analysis Day ----
+# See the "Derive/Impute Numeric Date/Time and Analysis Day" vignette section
+# for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#datetime)
+
+# Add questionnaire analysis date (ADT) and treatment start date (TRTSDT)
+adcoeq <- adcoeq %>%
+  derive_vars_dt(new_vars_prefix = "A", dtc = QSDTC) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT))
+
+# Derive visit info ----
+# See the "Visit and Period Variables" vignette for more information:
+# (https://pharmaverse.github.io/admiral/articles/visits_periods.html)
+
+# Derive analysis visit (AVISIT, AVISITN)
+adcoeq <- adcoeq %>%
+  mutate(
+    AVISIT = case_when(
+      is.na(VISIT) ~ NA_character_,
+      str_detect(VISIT, "UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      str_detect(VISIT, "SCREEN") ~ "SCREENING",
+      ADT <= TRTSDT ~ "Baseline",
```

Comment:
This might be a very stupid question, but would we not be able to use ```VISIT == "BASELINE" ~ "Baseline"```

And then, dependent on if we change in in the vignette, do the same there.

Reply:
ADT <= TRTSDT has been a standard when deriving baseline records. So I think we should include it and also add baseline check 
ADT <= TRTSDT & VISIT == "Baseline". This makes sense ?

Reply:
I would also rather kept it simple, so AVISIT = VISIT or AVISIT = str_to_title(VISIT)

Reply:
I agree with @starosto . Lets keep it as simple as possible. Then people are free to implement their own logic. I will make a issue to change this in the vignette to align. So feel free to change this @Siddhesh2097 . Then we can get this merged 👍